### PR TITLE
Fixes for the current running model

### PR DESCRIPTION
**Pull request by MayMeta**
This PR represent the last batch of changes, needed for the "reproduce_best_setup" model to run properly.

### CODE REVIEW

**Comment by MayMeta** [comment]
audio.py small-fix is not relevant anymore, due to a refactored resample function in master.

File: audio.py [hunk 1]
```diff
@@ -6,182 +6,182 @@
 import scipy.io.wavfile
 
 smax = torch.iinfo(torch.int16).max
-f2s_numpy = lambda signal, max=np.float32(smax): np.multiply(signal, max).astype('int16')
-s2f_numpy = lambda signal, max=np.float32(smax): np.divide(signal, max, dtype='float32')
+f2s_numpy = lambda signal, max = np.float32(smax): np.multiply(signal, max).astype('int16')
+s2f_numpy = lambda signal, max = np.float32(smax): np.divide(signal, max, dtype = 'float32')
```
Comment (FedorMeta) [inline audio.py:10]: This file only contains codestyle fixes.
Last Alexander Abugaliev commit in master did not follow the repository codestyle rules, which lead to merge conflicts.

File: tools.py [hunk 2]
```diff
@@ -160,24 +160,35 @@ def du(input_path):
 	)
 
 
-def csv2json(input_path, gz, group, reset_duration):
+def csv2json(input_path, gz, group, reset_begin_end, csv_sep):
+	""" Convert cvs transcripts file to .csv.json transcripts file. Each line in `input_path` file must have format:
+		"audio_path,transcription,begin,end\n"
+		csv_sep could be "comma", representing ",", or "tab", representing "\t".
+	"""
```
Comment (MayMeta) [inline tools.py:170]: This file contains csv2json tool refactoring. This tool is widely used in the project, in the process of creating the datasets. Each dataset uses .csv and .csv.json file format conventions.  I introduced a csv_sep parameter, which lets you control, which csv separator you want to use (tab by default).

File: tools.py [hunk 3]
```diff
@@ -160,24 +160,35 @@ def du(input_path):
...
 	gzopen = lambda file_path, mode = 'r': gzip.open(file_path, mode + 't') if file_path.endswith('.gz') else open(file_path, mode)
 
 	def duration(audio_name):
-		begin, end = map(float, os.path.splitext(audio_name)[0].split('_')[-2:])
-		return end - begin
+		begin, end = os.path.splitext(audio_name)[0].split('_')[-2:]
```
Comment (vadimkantorov) [inline tools.py]: Please add a range check. What's the format of youtube segment files?

File: tools.py [hunk 4]
```diff
@@ -160,24 +160,35 @@ def du(input_path):
...
+		return float(end) - float(begin)
```
Comment (FedorMeta) [inline tools.py:172]: Add strict check for audio_file_name format

File: tools.py [hunk 5]
```diff
@@ -160,24 +160,35 @@ def du(input_path):
...
-		dict(
-			audio_path = s[0],
-			ref = s[1],
-			begin = 0.0,
-			end = float(s[2]) if not reset_duration else duration(os.path.basename(s[0])),
-			**(dict(group = s[0].split('/')[group]) if group >= 0 else {})
-		) for l in gzopen(input_path) if '"' not in l for s in [l.strip().split(',')]
-	]
 	output_path = input_path + '.json' + ('.gz' if gz else '')
-	json.dump(transcript, gzopen(output_path, 'w'), ensure_ascii = False, indent = 2, sort_keys = True)
```
Comment (vadimkantorov) [inline tools.py:180]: Please use the newly merged `utils.open_maybe_gz`

File: transcribe.py [hunk 6]
```diff
@@ -245,13 +246,13 @@ def main(args):
...
-			output_lines.append(f'{audio_path}\t{hyp}\t{begin}\t{end}\n')
+		if args.output_csv:
+			output_lines.append(csv_sep.join((audio_path, hyp, str(begin), str(end))) + '\n')
 
 		print('Done: {:.02f} sec\n'.format(time.time() - tic))
 
-	if args.output_tsv:
-		with open(os.path.join(args.output_path, "transcripts.tsv"), 'w') as f:
+	if args.output_csv:
+		with open(os.path.join(args.output_path, "transcripts.csv"), 'w') as f:
```
Comment (vadimkantorov) [inline transcribe.py]: nit: single quotes

**Pull request merged**